### PR TITLE
updating the vue tutorial

### PR DESCRIPTION
**Pull request by thisisjofrank**

### CODE REVIEW

File: runtime/tutorials/how_to_with_npm/vue.md [hunk 1]
```diff
@@ -1,251 +1,269 @@
...
-```
+In your `package.json` file, update the `scripts` field to include the
+following:
 
-If we go to `localhost:8000/api`, we see:
-
-![json response of dinosaurs](../images/how-to/vue/api-response.png)
+```jsonc
+{
+  "tasks": {
```
Comment (thisisjofrank) [inline runtime/tutorials/how_to_with_npm/vue.md:122]:
```suggestion
  "scripts": {
```

File: runtime/tutorials/how_to_with_npm/vue.md [hunk 2]
```diff
@@ -1,251 +1,269 @@
...
+    "dev": "deno task dev:api & deno task dev:vite",
+    "dev:api": "deno run --allow-env --allow-net api/main.ts",
+    "dev:vite": "deno run -A npm:vite"
```
Comment (thisisjofrank) [inline runtime/tutorials/how_to_with_npm/vue.md:125]:
```suggestion
    "dev:vite": "deno run -A npm:vite",
```

File: runtime/tutorials/how_to_with_npm/vue.md [hunk 3]
```diff
@@ -1,251 +1,269 @@
...
+The `Dinosaurs` component will fetch the list of dinosaurs from the API we set
+up earlier and render them as links using the
+[`RouterLink` component from Vue Router](https://router.vuejs.org/guide/).
+(Because we are making a TypeScript project, don't forget to specify the
+`lang="ts"` attribute on the script tag.) Add the following code to the
+`Dinosaurs.vue` file:
 
-- importing `store`
-- rendering `store.dinosaur` in the HTML
+```tsx title="Dinosaurs.vue"
```
Comment (marvinhagemeister) [inline runtime/tutorials/how_to_with_npm/vue.md:226]:
````suggestion
```vue title="Dinosaurs.vue"
````

File: runtime/tutorials/how_to_with_npm/vue.md [hunk 4]
```diff
@@ -1,251 +1,269 @@
...
+This will set up a router with two routes: `/` and `/:dinosaur`. The `HomePage`
+component will be rendered at `/` and the `Dinosaur` component will be rendered
+at `/:dinosaur`.
 
-## Update Vue components
+Finally, you can delete all of the code in the `src/App.vue` file to and update
+it to include only the `<RouterView>` component:
 
-In `Dinosaurs.vue`, we'll do three things:
+```tsx title="App.vue"
```
Comment (marvinhagemeister) [inline runtime/tutorials/how_to_with_npm/vue.md:198]:
````suggestion
```vue title="App.vue"
````

File: runtime/tutorials/how_to_with_npm/vue.md [hunk 5]
```diff
@@ -1,251 +1,269 @@
...
+component. The `:to` attribute of the `RouterLink` component specifies the route
+to navigate to when the link is clicked, and the `:key` attribute is used to
+uniquely identify each dinosaur.
+
+#### The Homepage component
 
+The homepage will contain a heading and then it will render the `Dinosaurs`
+component. Add the following code to the `HomePage.vue` file:
+
+```tsx title="HomePage.vue"
```
Comment (marvinhagemeister) [inline runtime/tutorials/how_to_with_npm/vue.md:260]:
````suggestion
```vue title="HomePage.vue"
````

File: runtime/tutorials/how_to_with_npm/vue.md [hunk 6]
```diff
@@ -254,102 +272,77 @@ export default {
...
-
-```ts
-import { defineConfig } from "npm:vite@^3.1.3";
-import vue from "npm:@vitejs/plugin-vue@^3.2.39";
+First, we'll set up some types for the data we'll be fetching. Create a
+`types.ts` file in the `src` directory and add the following code:
 
-import "npm:vue@^3.2.39";
-import "npm:vue-router@4";
+```ts title="types.d.ts"
```
Comment (marvinhagemeister) [inline runtime/tutorials/how_to_with_npm/vue.md:290]:
````suggestion
```ts title="types.ts"
````

File: runtime/tutorials/how_to_with_npm/vue.md [hunk 7]
```diff
@@ -254,102 +272,77 @@ export default {
...
-});
+type ComponentData = {
+  dinosaurDetails: null | Dinosaur;
+};
 ```
 
-Next, let's create a folder named `router`. In it, let's create `index.ts`:
+Then update the `Dinosaur.vue` file:
+
+```tsx title="Dinosaur.vue"
```
Comment (marvinhagemeister) [inline runtime/tutorials/how_to_with_npm/vue.md:303]:
````suggestion
```vue title="Dinosaur.vue"
````

**Review by marvinhagemeister — APPROVED** [review]
LGTM, some minor formatting nits

**Pull request merged**